app/js/handlebarsUserTable.js

```diff
@@ -11,18 +11,103 @@ function fillUserTable(HBTemplate) {
             return result.json()
         })
         .then(function(result) {
-            let user_list = document.querySelector(".user_list")
-            user_list.innerHTML = ""
-
-            if (result.success) {
-                result.data.forEach(function(userData) {
-                    let html = template(userData)
-                    user_list.innerHTML += html
-                })
-            } else {
-                user_list.innerHTML = "Please contact Admin, user list unavailable"
+            updateDisplayedUsers(result, template)
+        })
+}
+
+/**
+ *this fills the handlebars template with users that have not been soft deleted
+ *
+ * @param apiResponse
+ * @param template
+ */
+function updateDisplayedUsers(apiResponse, template) {
+    let user_list = document.querySelector(".user_list")
+    user_list.innerHTML = ""
+    if (apiResponse.success) {
+        apiResponse.data.forEach(function(userData) {
+            if (userData.deleted === "0") {
+                let html = template(userData)
+                user_list.innerHTML += html
             }
         })
+        addEditEventListeners()
+        addDeleteEventListeners()
+    } else {
+        user_list.innerHTML = "Please contact Admin, user list unavailable"
+    }
+}
+
+/**
+ * fills handlebars template by passing in object and inserts into the score_list div
+ *
+ * @param HBTemplate the handlebars template
+ * @param ObjFunction the function that creates an object of all fields required in scores page
+ */
+function fillEditModal(HBTemplate, userInfo) {
+    let template = Handlebars.compile(HBTemplate)
+    let modal_content = document.querySelector("#modal-content")
+
+    modal_content.innerHTML = ""
+
+    if (userInfo.name && userInfo.email && userInfo.id) {
+        let html = template(userInfo)
+        modal_content.innerHTML += html
+    } else {
+        modal_content.innerHTML = "Please contact Admin, user list unavailable"
+    }
+}
+
+/**
+ * this adds the event listener to the edit button on creation of it the button
+ *
+ */
+function addEditEventListeners() {
+    let editButtons = document.querySelectorAll(".modalBtn")
+    editButtons.forEach(function (editButton) {
+        editButton.addEventListener('click', function (e) {
+            openDialog()
+            let userInfo = createObjectFromParentElement(e)
+            populateEditModal(userInfo)
+        })
+    })
+}
+
+/**
+ * Turns data from parent element (userTable handlebars template) into an object
+ *
+ */
+function createObjectFromParentElement(event){
```

Comment:
line 80 missing psr2, make sure you keep it consistent throughout if you are gonna do it with any other lines

Reply:
resolved